users/views.py

```diff
@@ -440,18 +448,32 @@ def change_username_api_view(request: Request):
 
 
 @api_view(["PATCH"])
-def update_profile_api_view(request: Request):
-    user = request.user
+def update_profile_api_view(request: Request) -> Response:
+    user: User = request.user
     serializer = UserUpdateProfileSerializer(user, data=request.data, partial=True)
     serializer.is_valid(raise_exception=True)
 
-    unsubscribe_tags = serializer.validated_data.get("unsubscribed_mailing_tags")
+    is_spam, _ = check_profile_update_for_spam(
+        user, cast(UserUpdateProfileSerializer, serializer)
+    )
+
+    if is_spam:
+        user.soft_delete()
+        user.save()
+        send_deactivation_email(user.email)
+        return Response(
+            data={
+                "message": "This bio seems to be spam. Please contact support@metaculus.com if you believe this was a mistake.",
+                "error_code": "SPAM_DETECTED",
+            },
+            status=status.HTTP_403_FORBIDDEN,
+        )
```

Comment:
I think this is the part that Hlib mentioned would be worth running in a Task instead of synchronous.

I think this will still be fast enough, so let's leave it as it is and change later only if necessary. But can you add a comment before line 461 to the effect of "this is done synchronously, if profile updating takes too long, move this to an async task instead"?